File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.java
class Solution {
    public int solution(int[][] dots) {
        int n = 10000;
        int m = -1;
        for(int i = 1; i < dots.length; i++){
            if(dots[0][0] != dots[i][0] && dots[0][1] != dots[i][1]) continue;
            n = Math.min(Math.abs(dots[0][0] - dots[i][0]) + Math.abs(dots[0][1] - dots[i][1]), n);
            m = Math.max(Math.abs(dots[0][0] - dots[i][0]) + Math.abs(dots[0][1] - dots[i][1]), m);
        }
        return n * m;
    }
}

File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.py
def solution(dots):
    xs, ys = set(), set()
    for dot in dots:
        xs.add(dot[0])
        ys.add(dot[1])
    xs, ys = list(xs), list(ys)
    return abs(xs[0] - xs[1]) * abs(ys[0] - ys[1])